services/navigation/store.go

```diff
@@ -119,7 +119,7 @@ func (store *MemoryNavigationStore) RemoveWaypoint(ctx context.Context, id primi
 	}
 	store.mu.Lock()
 	defer store.mu.Unlock()
-	newWps := make([]*Waypoint, 0, len(store.waypoints)-1)
+	newWps := []*Waypoint{}
```

Comment:
Fixes an off by one error where removing the last waypoint will panic: https://viam.atlassian.net/browse/RSDK-4934

Reply:
Would it not be more correct to use `make([]*Waypoint, 0, len(store.waypoints))` to fix the off-by-one?

Reply:
The thing is that the last parameter is the capacity, which sets how much memory is allocated for the slice. 

I don't think is necessary to set that until we know that we need the performance. 

If we do `make([]*Waypoint, 0, len(store.waypoints))`  then we will never reduce the capacity of the store for the entire lifetime of the nav service which is basically a memory leak. 

Creating a new slice every time we remove an element is slower but prevents leaking memory.

Reply:
Good point about the leak. Adding a million waypoints on a 64-bit cpu is 8mb, which is a solid chunk.

I think the right thing to do here would be to do a `make([]*Waypoint, 0, math.Max(0, len(store.waypoints)-1))`

The leak would only really be impactful if large numbers of waypoints are removed, in which case we want the performance.

Reply:
The capacity isn't immutable; if we try to append beyond the capacity in Go, the capacity will be increased under the hood to accommodate the new element. 

We're iterating the store regardless as we append. So if we set the capacity to len-1, and the wp isn't in the store, then there will be one extra allocation as the slice is expanded under the hood at our append.

Though, I just had another thought. Rewinding for a moment and looking again at `make([]*Waypoint, 0, len(store.waypoints))`, I don't think that will exhibit the memory leak you're talking about.

If we have a store of length 7, remove one, then the store is now a new array of capacity 7 and length 6. Remove another, and that new array has capacity 6, and length 5. Since we measure length but set capacity, the capacity *will* shrink over the lifetime of the service; it will always be one larger than the actual length.